pkg/service/oidc4vp/oidc4vp_service.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//go:generate mockgen -destination oidc4vp_service_mocks_test.go -self_package mocks -package oidc4vp_test -source=oidc4vp_service.go -mock_names transactionManager=MockTransactionManager,events=MockEvents,kmsRegistry=MockKMSRegistry,requestObjectPublicStore=MockRequestObjectPublicStore
+
+package oidc4vp
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jwt"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/presexch"
+
+	"github.com/trustbloc/vcs/pkg/doc/vc"
+	vcsverifiable "github.com/trustbloc/vcs/pkg/doc/verifiable"
+	vcskms "github.com/trustbloc/vcs/pkg/kms"
+	profileapi "github.com/trustbloc/vcs/pkg/profile"
+)
+
+var ErrDataNotFound = errors.New("data not found")
+
+type InteractionInfo struct {
+	AuthorizationRequest string
+	TxID                 TxID
+}
+
+type transactionManager interface {
+	CreateTransaction(presentationDefinition *presexch.PresentationDefinition) (*Transaction, string, error)
+}
+
+type events interface {
+	InteractionInitiated(txID TxID)
+	InteractionSucceed(txID TxID)
+}
+
+type requestObjectPublicStore interface {
+	Publish(requestObject string) (string, error)
+}
+
+type kmsRegistry interface {
+	GetKeyManager(config *vcskms.Config) (vcskms.VCSKeyManager, error)
+}
+
+type RequestObjectClaims struct {
+	VPToken VPToken `json:"vp_token"`
+}
+type VPToken struct {
+	PresentationDefinition *presexch.PresentationDefinition `json:"presentation_definition"`
+}
+
+type RequestObject struct {
```

Comment:
can we be more specific about what exactly `RequestObject` represents? is it `CredentialAuthorizationRequest`? `PresentationRequest`?

Reply:
Comment is added